package-res/resources/web/pentaho/type/instance.js

```diff
@@ -0,0 +1,274 @@
+     *
+     * require(["pentaho/type/Context"], function(Context) {
+     *   var context = new Context();
+     *   var Instance = context.get("instance");
+     *
+     *   var Derived = Instance.extend({
+     *     constructor: function(label) {
+     *       this.label = label;
+     *     },
+     *     type: { // type specification
+     *       greeting: "Hello, ",
+     *       veryLongString: "..."
+     *     },
+     *     saySomething: function() {
+     *       console.log(this.type.greeting + this.label + "!");
+     *     }
+     *   });
+     *
+     *   var a = new Derived("Alice");
+     *   a.saySomething(); // "Hello, Alice!"
+     *
+     *   var b = new Derived("Bob");
+     *   b.saySomething(); // "Hello, Bob!"
+     *
+     *   // All instances share the same _type_:
+     *   b.type.greeting ===  a.type.greeting // true
+     * });
+     *
+     * @description Creates an instance of this type.
+     *
+     * @see pentaho.type.spec.IInstance
+     * @see pentaho.type.spec.IInstanceProto
+     * @see pentaho.type.spec.ITypeProto
+     */
+    var Instance = Base.extend("pentaho.type.Instance", /** @lends pentaho.type.Instance# */{
+      // NOTE: not calling base to block default Base.js from copying 1st argument into `this`.
+      constructor: function() {
+      },
+
+      // Properties to ignore within extend
+      extend_exclude: {_: 1},
+
+      //region type property
+      _type: null, // Set on Type#_init
+
+      /**
+       * Gets the type of this instance.
+       *
+       * @type pentaho.type.Type
+       * @readonly
+       * @see pentaho.type.Instance.type
+       * @see pentaho.type.Type#instance
+       */
+      get type() {
+        return this._type;
+      },
+
+      // Supports Type instance-level configuration only. Can only be called on the prototype,
+      //  through Instance#implement!
+      // Not documented on purpose, to avoid users trying to configure a type
+      //  when they already have an instance of it, which is not supported...
+      // However, this is the simplest and cleanest way to implement:
+      //   Instance.implement({type: {.}})
+      // to mean
+      //   Type.implement({.}).instance.constructor
+      set type(config) {
+        // Class.implement essentially just calls Class#extend.
+        if(config) this.type.extend(config);
+      }, //endregion
+
+      //region serialization
+      /**
+       * Creates a top-level specification that describes this instance.
+       *
+       * If an [ambient specification context]{@link pentaho.type.SpecificationContext.current},
+       * currently exists, it is used to manage the serialization process.
+       * Otherwise, one is created and set as current.
+       * Then, the actual work is delegated to {@link pentaho.type.Instance#toSpecInContext}.
+       *
+       * @param {Object} [keyArgs] - The keyword arguments object.
+       * Passed to every instance and type serialized within this scope.
```

Comment:
Suggest rewording into a sentence structure:

The keyword arguments object is passed to every type and instance serialized within this scope.